From 943ff31da1bbf3954c4e9f223b2049da321e7398 Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Tue, 15 May 2018 16:06:23 +0800
Subject: [PATCH] arm64: dts: rockchip: using multicodec for RK3308 EVB

If we need to extend custom audio features, using multicodec
is much more convenient than limited simple-card.

Change-Id: I784e38d33ded9a7dd9c253595611f343a4dcd566
Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
---
 .../boot/dts/rockchip/rk3308-evb-v10.dtsi     | 19 ++++++-------------
 .../boot/dts/rockchip/rk3308-evb-v11.dtsi     | 19 ++++++-------------
 2 files changed, 12 insertions(+), 26 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
index fb3b8c86df5e..f4bceb77b8b4 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
@@ -103,19 +103,12 @@
 	};
 
 	acodec_sound: acodec-sound {
-		compatible = "simple-audio-card";
-		simple-audio-card,format = "i2s";
-		simple-audio-card,name = "rockchip,rk3308-acodec";
-		simple-audio-card,mclk-fs = <256>;
-		simple-audio-card,codec-hp-det;
-		simple-audio-card,widgets =
-			"Headphone", "Headphones";
-		simple-audio-card,cpu {
-			sound-dai = <&i2s_8ch_2>;
-		};
-		simple-audio-card,codec {
-			sound-dai = <&acodec>;
-		};
+		compatible = "rockchip,multicodecs-card";
+		rockchip,card-name = "rockchip,rk3308-acodec";
+		rockchip,codec-hp-det;
+		rockchip,mclk-fs = <256>;
+		rockchip,cpu = <&i2s_8ch_2>;
+		rockchip,codec = <&acodec>;
 	};
 
 	spdif_tx_sound: spdif-tx-sound {
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-v11.dtsi b/arch/arm64/boot/dts/rockchip/rk3308-evb-v11.dtsi
index 3947350c7caa..d54f53907e84 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-v11.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-v11.dtsi
@@ -88,19 +88,12 @@
 	};
 
 	acodec_sound: acodec-sound {
-		compatible = "simple-audio-card";
-		simple-audio-card,format = "i2s";
-		simple-audio-card,name = "rockchip,rk3308-acodec";
-		simple-audio-card,mclk-fs = <256>;
-		simple-audio-card,codec-hp-det;
-		simple-audio-card,widgets =
-			"Headphone", "Headphones";
-		simple-audio-card,cpu {
-			sound-dai = <&i2s_8ch_2>;
-		};
-		simple-audio-card,codec {
-			sound-dai = <&acodec>;
-		};
+		compatible = "rockchip,multicodecs-card";
+		rockchip,card-name = "rockchip,rk3308-acodec";
+		rockchip,codec-hp-det;
+		rockchip,mclk-fs = <256>;
+		rockchip,cpu = <&i2s_8ch_2>;
+		rockchip,codec = <&acodec>;
 	};
 
 	spdif_tx_sound: spdif-tx-sound {
-- 
2.35.3

